src/util/json.h

```diff
@@ -2,14 +2,147 @@
 //  Chair of Algorithms and Data Structures.
 //  Author: Johannes Kalmbach <kalmbach@cs.uni-freiburg.de>
 
-// Convenience header for Nlohmann::Json that sets the default options.
+// Convenience header for Nlohmann::Json that sets the default options. Also
+// adds support for std::optional.
 
 #ifndef QLEVER_JSON_H
 #define QLEVER_JSON_H
 
 // Disallow implicit conversions from nlohmann::json to other types,
 // most notably to std::string.
 #define JSON_USE_IMPLICIT_CONVERSIONS 0
+
 #include <nlohmann/json.hpp>
+#include <optional>
+#include <variant>
+#include <utility>
+#include <concepts>
+
+// Added support for serializing `std::optional` using `nlohmann::json`.
+namespace nlohmann {
+    template <typename T>
+    struct adl_serializer<std::optional<T>> {
+        static void to_json(nlohmann::json& j, const std::optional<T>& opt) {
+          if (opt.has_value()) {
+            j = opt.value();
+          } else {
+            j = nullptr;
+          }
+        }
+
+        static void from_json(const nlohmann::json& j, std::optional<T>& opt) {
+          if (j.is_null()){
+            opt = std::nullopt;
+          } else {
+            opt = j.get<T>();
+          }
+        }
+    };
+}
+
+// Added support for serializing `std::monostate` using `nlohmann::json`.
+namespace nlohmann {
+    template<>
+    struct adl_serializer<std::monostate>{
+        static void to_json(nlohmann::json& j,
+            const std::monostate&) {
+          // Monostate is just an empty placeholder value.
+          j = nullptr;
+        }
+
+        static void from_json(const nlohmann::json&,
+            std::monostate&) {
+          // Monostate holds no values, so the given monostate is already
+          // identical to the serialized one.
+          // TODO Add an exception here, if the j doesn't contain a nullptr.
+          // I mean, there is literally no sense in trying to convert any actual
```

Comment:
I made one, but it's a bit rough, because I don't know, where to find some of the arguments. See [c8f23ff](https://github.com/ad-freiburg/qlever/pull/860/commits/c8f23ff81fe8b78c61138d754ffcaf242e3958ef).

Reply:
https://json.nlohmann.me/home/exceptions/#jsonexceptiontype_error302

Reply:
Done, see [c5ec08c](https://github.com/ad-freiburg/qlever/pull/860/commits/c5ec08c5ce4d987825a23149257b091d117a034f).